### ISSUE TITLE
Cut `R CMD check` times down to < 10  minutes

### ISSUE THREAD

**Issue by gowerc** [Genentech/jmpost#264]
This is not a priority issue as we don't currently have any plans to publish to CRAN (at least we can't until cmdstanr does) but if we ever did want to publish we can't unless the total run time of our vignettes and unit-tests is < 10 minutes. 

Main issue at the moment is a fresh run on CRAN wouldn't have access to any cached models so would take a long time to run. Might need to be quite selective on which tests we enable on CRAN.

**Comment by danielinteractive** [comment]
Yep, and can also learn from `psborrow2` for these things because it uses a similar framework. I think they are going for caching / pre-compiling quite a few things

**Comment by danielinteractive** [comment]
yeah you can sync with @gravesti and @mattsecrest on the approaches

**Comment by gravesti** [comment]
In psborrow we are doing something like this: https://ropensci.org/blog/2019/12/08/precompute-vignettes/
That is we knit the Rmd but don't run render to html or pdf, so the tables and figures are generated in the markdown and stored in the repo.
This should allow us to get on CRAN even without cmdstanr.
I have wrapped all the checks and tests that need sampling (results) in if statements to check if cmdstan and cmdstanr are available.
That means we can still have examples and tests for all of the stuff which sets up the models.
